### PR TITLE
fix: preserve typed wall-clock times on event save

### PR DESCRIPTION
**Pull request by shayani**
The event form silently dropped user-typed start/end times whenever the typed local
hour numerically matched the stored UTC hour (e.g. typing 12:00 on a Brussels event
whose stored UTC hour was already 12). The user could change times, save, and see
the same old times still displayed.

After this fix, whatever start/end hour the user types in the form is preserved as
the event's local time, regardless of time zone.

<h3>Greptile Summary</h3>

This PR fixes a silent time-preservation bug where typed local hours were dropped on save whenever the wall-clock hour numerically matched the stored UTC hour. The fix introduces a `TimeZone::WallClockToUtc` service that reads only the wall-clock components of the submitted datetime and converts them using the event's IANA time zone, with deterministic handling of DST spring-forward gaps (`TZInfo::PeriodNotFound` → shift +1 h) and fall-back overlaps (`TZInfo::AmbiguousTime` → pick DST/summer occurrence). A `dates_from_form` transient flag is set by the controller in both `create` and `update` to activate this path only when the user submitted a typed wall-clock value.

<h3>Confidence Score: 5/5</h3>

Safe to merge — correct DST edge-case handling with TZInfo 2.0.6 API, no unguarded exception paths on normal inputs, and comprehensive test coverage.

No P0 or P1 findings. The service correctly catches TZInfo::AmbiguousTime and TZInfo::PeriodNotFound (the two TZInfo 2.x exceptions for DST boundaries). The dates_from_form flag is set before save in both create and update. The rescue ActionView::Template::Error block in update also benefits from the already-set flag on @event. Tests cover all advertised scenarios.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| app/services/time_zone/wall_clock_to_utc.rb | New service that correctly converts wall-clock datetime components to UTC, handling TZInfo::AmbiguousTime (fall-back overlap) and TZInfo::PeriodNotFound (spring-forward gap) with deterministic policies; uses TZInfo 2.0.6-compatible API. |
| app/models/event.rb | Replaces date_start_changed? guard with dates_from_form flag check; delegates conversion to WallClockToUtc service; correctly conditions on both date_start and date_end being present. |
| app/controllers/events_controller.rb | Adds mark_dates_from_form helper called at the top of both create and update, correctly gating on params[:time_start].present? before setting the transient flag. |
| test/models/event/dates_from_form_test.rb | Comprehensive integration tests covering the collision bug, timezone-change-during-save, no-drift round-trip (with documented UTC+2 assumption), spring-forward gap, fall-back overlap, and no-flag skip path. |
| test/services/time_zone/wall_clock_to_utc_test.rb | Unit tests for the new service; cover non-ambiguous conversion, DST fall-back (picks CEST occurrence at 00:30 UTC), spring-forward gap (shifts to 01:30 UTC), and zone-stripped input. |

</details>

</details>

<sub>Reviews (2): Last reviewed commit: ["refactor: extract wall-clock-to-UTC conv..."](https://github.com/eventaservo/eventaservo/commit/8abb95b69ec5c6ca75524b1f1d98673d33c7ddbb) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=29703491)</sub>

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -495,7 +502,7 @@ def format_event_data
     result = TimeZone::Normalize.call(time_zone)
     self.time_zone = result.success? ? result.payload : "Etc/UTC"
 
-    if date_start_changed?
+    if dates_from_form && date_start.present? && date_end.present?
       tz = TZInfo::Timezone.get(time_zone)
       self.date_start = tz.local_to_utc(Time.new(date_start.year, date_start.month, date_start.day, date_start.hour, date_start.min)).in_time_zone(time_zone)
       self.date_end = tz.local_to_utc(Time.new(date_end.year, date_end.month, date_end.day, date_end.hour, date_end.min)).in_time_zone(time_zone)
```
[the remark this answers is not in the document]
Comment (shayani) [reply app/models/event.rb]: Addressed in 14858919: wrapped `local_to_utc` in a `wall_clock_to_utc` helper that rescues `TZInfo::AmbiguousTime` (picks the DST occurrence) and `TZInfo::PeriodNotFound` (shifts +1h). New tests cover spring-forward and fall-back boundaries on `Europe/Brussels`.

File: test/models/event/dates_from_form_test.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    event.dates_from_form = true
+    event.assign_attributes(
+      date_start: "2026-05-16 14:00",
+      date_end: "2026-05-16 19:00"
+    )
+    event.save!
+
+    assert_equal Time.utc(2026, 5, 16, 12, 0), event.date_start.utc
+    assert_equal Time.utc(2026, 5, 16, 17, 0), event.date_end.utc
+  end
```
[the remark this answers is not in the document]
Comment (shayani) [reply test/models/event/dates_from_form_test.rb]: Addressed in 14858919: renamed the test to "round-trips the local wall-clock time the form re-renders without drift", added an explicit local-representation assertion, and documented the May 2026 CEST offset assumption in a comment. The implicit timing dependency is now visible.

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -148,6 +151,7 @@ def update # rubocop:disable Metrics/MethodLength, Metrics/PerceivedComplexity,
   rescue ActionView::Template::Error => e
     Sentry.capture_exception(e)
     PaperTrail.request.disable_model(Event)
+    mark_dates_from_form
```
[the remark this answers is not in the document]
Comment (shayani) [reply app/controllers/events_controller.rb]: Addressed in 14858919: removed the redundant call from the rescue block. The first `mark_dates_from_form` at the top of `update` is enough.
